pkg/actuators/machine/utils_test.go

```diff
@@ -54,6 +56,32 @@ func TestExtractNodeAddresses(t *testing.T) {
 				{Type: corev1.NodeInternalDNS, Address: "ec2.example.net"},
 				{Type: corev1.NodeHostName, Address: "ec2.example.net"},
 			},
+			domainNames: nil,
+		},
+		{
+			testcase: "custom-domain",
+			instance: &ec2.Instance{
+				PrivateDnsName: aws.String("ec2.example.net"),
+				NetworkInterfaces: []*ec2.InstanceNetworkInterface{
+					{
+						Status: aws.String(ec2.NetworkInterfaceStatusInUse),
+						PrivateIpAddresses: []*ec2.InstancePrivateIpAddress{
+							{
+								Primary:          aws.Bool(true),
+								PrivateIpAddress: aws.String("10.0.0.5"),
+							},
+						},
+					},
+				},
+			},
+			expectedAddresses: []corev1.NodeAddress{
```

Comment:
It'd be nice to include a case that checks we don't duplicate as per https://github.com/openshift/cluster-api-provider-aws/pull/318/files#diff-9d8c56b0090958d82aa8280b211d4bf8R309-R311

Reply:
Done.